### ISSUE TITLE
Doesn't seem to support typescript methods

### ISSUE THREAD

**Issue by gwynjudd** [Microsoft/vscode-comment#1]
e.g., in this class:

```
    export class TwoPageView extends Marionette.LayoutView<TwoPageModel> {
        private __leftContent: Content.ContentController = null;
        private __leftView: Page.PageView;

        private __rightContent: Content.ContentController = null;
        private __rightView: Page.PageView;

        public initialize() {
            logger.trace("initialize");
            this.listenTo(this, "view.updateParentController", this.__onSetParentController);
            this.listenTo(this.model, "change:split", this.__onSetPageSplit);

            this.addRegions({
                'leftRegion': '.js-region-left',
                'rightRegion': '.js-region-right'
            });
        }
    }
```

You can't add doc comments for "public initialize()" - no matter what I try it always says "Please select a TypeScript or JavaScript function signature"

![image](https://cloud.githubusercontent.com/assets/314188/11251387/dac39bf6-8e95-11e5-9265-27fe6bd6fb61.png)

**Comment by stevencl** [comment]
Thanks, I'll look into this. My implementation is quite naïve...

**Comment by stevencl** [comment]
I had some very simple code that was just looking for the presence of the function keyword in the selected line. I have removed this check so it should work for the case above. I pushed the changes and updated the extension.

As a result though, if the user selects any line of code and runs the extension the code will attempt to generate a comment for that line even though it may not contain a function signature, which isn't great. I can't think of a better way to check for this though without properly parsing the file and building up a syntax tree.

I'd be more than happy to take pull requests :-)
